alica_engine/include/engine/BasicBehaviour.h

```diff
@@ -78,7 +78,10 @@ class BasicBehaviour
     bool isEventDriven() const { return _behaviour->isEventDriven(); }
 
     // This is not thread safe. Should only be called by the scheduler thread. TODO: make this private
-    std::optional<std::string> getTraceContext() const;
+    std::optional<std::string> getTraceContext() const
+    {
+        return _trace ? std::optional<std::string>(_trace->context()) : std::nullopt;
+    }
```

Comment:
Will this still go into 0.9.2? If this goes into rr_devel, getTraceContext has been moved to RunnableObject there.

Reply:
yes this is for 0.9.2. I'll make a separate PR for rr_devel